packages/context-menu/src/vaadin-contextmenu-event.js

```diff
@@ -34,8 +34,24 @@ register({
     }
   },
 
-  touchstart(e) {
+  _setSourceEvent(e) {
     this.info.sourceEvent = e;
+
+    const path = e.composedPath();
+
+    // Calling `sourceEvent.composedPath()` after a timeout would return an empty array.
+    // This is especially problematic on iOS where we configure the timer on touchstart.
+    // So we need this hack to make `grid.getEventContext(sourceEvent)` work properly.
+    Object.defineProperty(this.info.sourceEvent, 'composedPath', {
```

Comment:
Instead of patching a native function from the event, would it be safer to add a new private property to it? 

Something like `sourceEvent.__composedPath = path`. Then `getEventContext` could primarily look for that property `const path = event__composedPath || event.composedPath();`

Reply:
Updated the PR and added corresponding logic to `vaadin-grid` to ensure `getEventContext()` works.
I have no idea how to test this without using `__composedPath` in tests also 😕

Reply:
I guess there's no way to test it since we can't force vaadin-contextmenu-event to use iOS in a test case